homeassistant/components/smscom/notify.py

```diff
@@ -0,0 +1,121 @@
+"""SMS COM platform for notify component."""
+import logging
+import time
+
+from homeassistant.components.notify import (ATTR_TARGET, PLATFORM_SCHEMA,
+                                             BaseNotificationService)
+from homeassistant.const import CONF_RECIPIENT
+import homeassistant.helpers.config_validation as cv
+from serial import Serial
+import voluptuous as vol
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PORT = "com_port"
+DEFAULT_PORT = "/dev/ttyUSB0"
+CONF_BAUDRATE = "com_baudrate"
+DEFAULT_BAUDRATE = 115200
+
+PLATFORM_SCHEMA = vol.Schema(
+    vol.All(
+        PLATFORM_SCHEMA.extend(
+            {
+                vol.Optional(CONF_BAUDRATE, default=DEFAULT_BAUDRATE): cv.positive_int,
+                vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.string,
+                vol.Required(CONF_RECIPIENT, default=[]): vol.All(
+                    cv.ensure_list, [cv.string]
+                ),
+            }
+        )
+    )
+)
+
+serial = Serial()
+
+def get_service(hass, config, discovery_info=None):
+    """Get the atSMS notification service."""
+
+    serial.port = config[CONF_PORT]
+    serial.baudrate = config[CONF_BAUDRATE]
+    serial.timeout = 1
+
+    if not _check_com_port(config):
+        _LOGGER.error("Can't access to %s", config[CONF_PORT])
+        return None
+    return SMSComNotificationService(config)
+
+class StaticWait:
+    """Wait sendding by static variable"""
+    still_sending = False
+
+class SMSComNotificationService(BaseNotificationService):
+    """Implementation of a notification service for the SMS Com service."""
+
+    def __init__(self, config):
+        """Initialize the service."""
+        self.port = config[CONF_PORT]
+        self.recipients = config[CONF_RECIPIENT]
+
+    def send_message(self, message="", **kwargs):
+        """Send a message to a user."""
+        phones = kwargs.get(ATTR_TARGET, self.recipients)
+        while StaticWait.still_sending:
+            time.sleep(1)
+
+        StaticWait.still_sending = True
+        for phone in phones:
+            _LOGGER.debug("Send message to:\r\n%s", phone)
+            if not serial.is_open:
+                serial.open()
+            serial.write(b'AT+CMGF=1\r\n')
```

Comment:
API specific code and communication with devices should be part of a third party library hosted on PyPi.

See development documentation with tutorial how to publish a PyPi package.

https://developers.home-assistant.io/docs/creating_component_code_review/#4-communication-with-devicesservices